lib/main/atom/commands/goToDeclaration.ts

```diff
@@ -12,6 +21,8 @@ commands.set("typescript:go-to-declaration", deps => {
     const client = await deps.getClient(location.file)
     const result = await client.executeDefinition(location)
 
+    prevCursorPositions.push(location);
```

Comment:
It's possible that the user executes this command, gets multiple results, but presses escape and doesn't open any of them. We don't want to record the previous position in that case - only if they make the jump.